### DISCUSSION TITLE
Can't update a nullable value to null

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2290] [Q&A] [answered]
I have a row with a nullable int column. At first it has an int and all is fine, but when I try to do `update` with 
`row.copyWith(myCol: Value(null))` nothing happens... it's like I put `Value.absent()`. If I put any other int it updates fine.

am I doing something wrong or is this expected behavior? is there a workaround?

**Comment by simolus3** [comment, marked as answer]
I see how this is kind of confusing, but it's also expected.

1. The `copyWith` will set the field back to null for the created row object.
2. But then, for the update, drift still doesn't know whether `null` on the row object means that the value should be set to null or kept as it is.

For reason 2, I recommend using the generated companion classes if you need inserts or updates. You can also do an update with `row.copyWith(myCol: Value(null)).toCompanion(false)`. The `false` in `toCompanion` means that `null`s should explicitly be treated as present.

**Reply by BananaMasterz** [reply]
ah yes thank you... didn't think about it the right way. thanks!
